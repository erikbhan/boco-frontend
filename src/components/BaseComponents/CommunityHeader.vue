<template>
  <div class="flex items-center justify-between mx-4">
    <div class="flex-1 min-w-0">
      <h2
        class="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate"
      >
        {{ community.name }}
      </h2>
      <div
        class="mt-1 flex flex-col sm:flex-row sm:flex-wrap sm:mt-0 sm:space-x-6"
      >
        <div class="mt-2 flex items-center text-sm text-gray-500">
          <svg
            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 20 20"
            fill="currentColor"
            aria-hidden="true"
          >
            <path
              fill-rule="evenodd"
              d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
              clip-rule="evenodd"
            />
          </svg>
          {{ community.location }}
        </div>
      </div>
    </div>
    <div>
      <!-- If the user is not a member in the community, this button will show -->
      <div v-if="!member">
        <ColoredButton
            v-if="!member"
            :text="'Bli med'"
            @click="joinCommunity(community.communityId)"
            class="m-2"
        />

      <CustomFooterModal
          @close="this.dialogOpen = false"
          :visible="dialogOpen"
          title="Kan ikke bli med"
          message="Logg inn først for å bli med i en gruppe."
      />
      </div>

      <!-- If the user is member of the community, this hamburger menu will show -->
      <div v-if="member">
        <svg @click="toggle" xmlns="http://www.w3.org/2000/svg" class="w-9 h-9 cursor-pointer" fill="none" viewBox="0 0 24 24" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h16M4 18h16" />
      </svg>

      <CommunityHamburger v-if="hamburgerOpen" class="origin-top-right absolute right-0" :community-i-d="community.communityId"/> <!-- class="absolute" -->
      </div>
    </div>
  </div>
</template>

<script>
import CommunityHamburger from "@/components/CommunityComponents/CommunityHamburger";
import ColoredButton from "@/components/BaseComponents/ColoredButton";
import { JoinOpenCommunity, GetIfUserAlreadyInCommunity } from "@/utils/apiutil";
import CustomFooterModal from "@/components/BaseComponents/CustomFooterModal";

export default {
  name: "CommunityHeader",
  components: {
    CommunityHamburger,
    ColoredButton,
    CustomFooterModal,
  },
  data(){
    return{
      hamburgerOpen: false,
      dialogOpen: false,
      member: true,
    }
  },
  props: {
    adminStatus: Boolean,
    community: {
      communityId: Number,
      name: String,
      description: String,
      visibility: Number,
      location: String,
      picture: String,
    }
  },
  methods: {
    //To open and close the hamburger menu
    toggle: function (){
      if(this.hamburgerOpen){
        this.hamburgerOpen = false;
      }
      else{
        this.hamburgerOpen = true;
      }
    },
    joinCommunity: async function(id){
      const response = await JoinOpenCommunity(id);
      if(response === "Login to join any community"){
        this.dialogOpen = true;
      }
      else{
        window.location.reload();
      }
    },
    getIfUserInCommunity: async function(){
      try{
        this.member = await GetIfUserAlreadyInCommunity(this.$router.currentRoute.value.params.communityID);
      } catch (error){
        console.log(error);
      }
    }
  },
  beforeMount() {
    this.getIfUserInCommunity();
  }
};
</script>
