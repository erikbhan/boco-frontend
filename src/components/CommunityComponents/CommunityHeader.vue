<template>
  <div class="flex items-center justify-between mx-4">
    <router-link
      :to="'/community/' + community.communityId"
      class="flex-1 min-w-0"
    >
      <h2
        class="text-xl md:text-2xl text-gray-600 font-medium w-full sm:truncate"
      >
        {{ community.name }}
      </h2>
      <div
        class="mt-1 flex flex-col sm:flex-row sm:flex-wrap sm:mt-0 sm:space-x-6"
      >
        <div class="mt-2 flex items-center text-sm text-gray-500">
          <svg
            class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 20 20"
            fill="currentColor"
            aria-hidden="true"
          >
            <path
              fill-rule="evenodd"
              d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
              clip-rule="evenodd"
            />
          </svg>
          {{ community.location }}
        </div>
      </div>
    </router-link>
    <div>
      <!-- If the user is not a member in the community, this button will show -->
      <div v-if="!member">
        <ColoredButton
          v-if="!member"
          :text="'Bli med'"
          @click="joinCommunity(community.communityId)"
          class="m-2"
        />

        <CustomFooterModal
          @close="this.dialogOpen = false"
          :visible="dialogOpen"
          title="Kan ikke bli med"
          message="Logg inn først for å bli med i en gruppe."
        />
      </div>

      <!-- If the user is member of the community, this hamburger menu will show -->
      <div v-if="member">
        <svg
          @click="toggleHamburgerMenu()"
          xmlns="http://www.w3.org/2000/svg"
          class="w-9 h-9 cursor-pointer"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
        >
          <path
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            d="M4 6h16M4 12h16M4 18h16"
          />
        </svg>

        <CommunityHamburger
          v-if="hamburgerOpen"
          class="origin-top-right absolute right-0"
          :community-i-d="community.communityId"
          :admin="admin"
        />
        <!-- class="absolute" -->
      </div>
    </div>
  </div>
</template>

<script>
import CommunityHamburger from "@/components/CommunityComponents/CommunityHamburger";
import ColoredButton from "@/components/BaseComponents/ColoredButton";
import CommunityService from "@/services/community.service";
import CustomFooterModal from "@/components/BaseComponents/CustomFooterModal";
import {
  JoinOpenCommunity,
  GetIfUserAlreadyInCommunity,
} from "@/utils/apiutil";

export default {
  name: "CommunityHeader",
  components: {
    CommunityHamburger,
    ColoredButton,
    CustomFooterModal,
  },
  data() {
    return {
      hamburgerOpen: false,
      dialogOpen: false,
      member: true,
      community: {},
    };
  },
  props: {
    admin: Boolean,
  },
  methods: {
    toggleHamburgerMenu() {
      this.hamburgerOpen = !this.hamburgerOpen;
    },
    joinCommunity: async function (id) {
      const response = await JoinOpenCommunity(id);
      if (response === "Login to join any community") {
        this.dialogOpen = true;
      } else {
        window.location.reload();
      }
    },
    getIfUserInCommunity: async function () {
      try {
        this.member = await GetIfUserAlreadyInCommunity(
          this.$router.currentRoute.value.params.communityID
        );
      } catch (error) {
        console.log(error);
      }
    },
  },
  beforeMount() {
    this.getIfUserInCommunity();
  },
  async mounted() {
    this.community = await CommunityService.getCommunity(
      this.$route.params.communityID
    );
  },
};
</script>
