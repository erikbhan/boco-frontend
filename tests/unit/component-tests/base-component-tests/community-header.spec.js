import { mount } from "@vue/test-utils";
import CommunityHeader from "@/components/BaseComponents/CommunityHeader.vue"

describe("CommunityHeader component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(CommunityHeader, {
        //passing prop to component
        props: {
            adminStatus: true,
            community: {
                communityId: 1,
                name: "String",
                description: "String",
                visibility: 0,
                location: "String",
                picture: "String",
            },
        }
    });
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});