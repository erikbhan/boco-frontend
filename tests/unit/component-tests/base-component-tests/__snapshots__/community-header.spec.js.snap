// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityHeader component renders correctly 1`] = `
<div
  class="flex items-center justify-between mx-4"
>
  <div
    class="flex-1 min-w-0"
  >
    <h2
      class="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate"
    >
      String
    </h2>
    <div
      class="mt-1 flex flex-col sm:flex-row sm:flex-wrap sm:mt-0 sm:space-x-6"
    >
      <div
        class="mt-2 flex items-center text-sm text-gray-500"
      >
        <svg
          aria-hidden="true"
          class="flex-shrink-0 mr-1.5 h-5 w-5 text-gray-400"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
            fill-rule="evenodd"
          />
        </svg>
         String
      </div>
    </div>
  </div>
  <div>
    <span
      class="hidden sm:block"
    >
      <!-- Legg dette til i button: v-if="adminStatus" -->
      <svg
        class="w-9 h-9 cursor-pointer"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <!--v-if-->
      <!-- class="absolute" -->
    </span>
  </div>
</div>
`;
